File: frontend/src/lib/types/activity.ts
export type ActivitySource =
    | "github_event"
    | "finding_detected"
    | "finding_resolved"
    | "health_changed"
    | "bot_interaction";

export interface BotInteractionDetails {
    intent: string;
    status: string;
    skip_reason: string | null;
    response_text: string | null;
    requester_github_login: string;
    question_text: string;
}

export interface ActivityItem {
    source: ActivitySource;
    timestamp: string;
    title: string;
    summary: string;
    reference_id: string;
    details?: BotInteractionDetails | null;
}

export interface ActivityFeedResponse {
    items: ActivityItem[];
    next_before: string | null;
}

File: backend/app/schemas/activity.py
import uuid
from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class ActivitySource(str, Enum):
    github_event = "github_event"
    finding_detected = "finding_detected"
    finding_resolved = "finding_resolved"
    health_changed = "health_changed"
    bot_interaction = "bot_interaction"


class ActivityItem(BaseModel):
    source: ActivitySource
    timestamp: datetime
    title: str
    summary: str
    reference_id: uuid.UUID
    details: dict | None = None


class ActivityFeedResponse(BaseModel):
    items: list[ActivityItem]
    next_before: datetime | None = None